core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1726,6 +1729,10 @@ abstract class RDD[T: ClassTag](
             // checkpoint ourselves
             dependencies.foreach(_.rdd.doCheckpoint())
           }
+          if (storageLevel == StorageLevel.NONE) {
+            logInfo(s"do checkpoint on unpersisted RDD ${id}, it will cause RDD recomputation" +
+              " when saving checkpoint files.")
```

Comment:
Can you please change the message to something like :
`s"checkpoint on unpersisted RDD $this will result in recomputation"`